RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -366,22 +372,30 @@ void HGCalCLUEAlgoT<T, STRATEGY>::calculateDistanceToHigher(const T& lt, const u
         // loop over all hits in this bin
         for (unsigned int j = 0; j < binSize; j++) {
           unsigned int otherId = lt[binId][j];
-          float dist = distance(lt, i, otherId, layerId);
+          float dist = distance2(lt, i, otherId, layerId);
           bool foundHigher =
               (cellsOnLayer.rho[otherId] > cellsOnLayer.rho[i]) ||
               (cellsOnLayer.rho[otherId] == cellsOnLayer.rho[i] && cellsOnLayer.detid[otherId] > cellsOnLayer.detid[i]);
-          if (foundHigher && dist <= i_delta) {
-            // update i_delta
+          if (foundHigher && dist < i_delta) {
+            rho_max = cellsOnLayer.rho[otherId];
+            i_delta = dist;
+            i_nearestHigher = otherId;
+          } else if (foundHigher && dist == i_delta && cellsOnLayer.rho[otherId] > rho_max) {
+            rho_max = cellsOnLayer.rho[otherId];
+            i_delta = dist;
+            i_nearestHigher = otherId;
+          } else if (foundHigher && dist == i_delta && cellsOnLayer.rho[otherId] == rho_max &&
+                     cellsOnLayer.detid[otherId] > cellsOnLayer.detid[i]) {
+            rho_max = cellsOnLayer.rho[otherId];
             i_delta = dist;
-            // update i_nearestHigher
             i_nearestHigher = otherId;
           }
         }
       }
     }
     bool foundNearestHigherInSearchBox = (i_delta != maxDelta);
     if (foundNearestHigherInSearchBox) {
-      cellsOnLayer.delta[i] = i_delta;
+      cellsOnLayer.delta[i] = std::sqrt(i_delta);
```

Comment:
I'm a bit confused, just a few lines here you have:
`cellsOnLayer.delta[i] = maxDelta;`
How come that one doesn't change as well?

Reply:
I'm not sure I understand. The other `delta` is inside the `else` block of this `if`.

Reply:
IIUC the comment, because `maxDelta` is just `std::numeric_limits<float>::max()`, _i.e._ a constant to indicate that no valid hit has been found (yet).